src/main/java/com/ibm/cloud/sdk/core/util/CredentialUtils.java

```diff
@@ -302,6 +313,41 @@ protected static VcapService getVcapServiceEntry(String serviceName) {
     return Collections.emptyMap();
   }
 
+  /**
+   * Returns a Map containing properties found within the process' system properties
+   * that are associated with the specified cloud service.
+   *
+   * @param serviceName the name of the cloud service whose properties should be
+   *                    retrieved
+   * @return a Map containing the properties
+   */
+  static Map<String, String> getSystemPropsCredentialsAsMap(String serviceName) {
+    // Retrieve the Map of environment variables from the current process.
+    Properties systemProperties = System.getProperties();
+
+    // Extract the properties related to the specified service and populate the
+    // result Map.
+    if (systemProperties != null) {
+      Map<String, String> props = new HashMap<>();
+      serviceName = serviceName.toUpperCase().replaceAll("-", "_") + "_";
+
+      for (Map.Entry<Object, Object> entry : systemProperties.entrySet()) {
+        String key = entry.getKey() == null ? "" : String.valueOf(entry.getKey()).trim();
```

Comment:
I think it would be ok to just check if the entry's key or value is null, and if so, just "continue" (i.e. skip that entry altogether).